src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2021,9 +2021,22 @@ private static BigInteger Subtract(uint[]? leftBits, int leftSign, uint[]? right
                 }
 
                 NumericsHelpers.DangerousMakeTwosComplement(xd); // Mutates xd
-                if (xd[^1] == 0)
+
+                if (smallShift == 0 && xd[0] == 1)
```

Comment:
It might just be that I'm not awake enough, but why aren't we simply doing `trackSignBit = (_sign & int.MinValue) != 0`?

That is, if the most significant bit of `_sign` is set, we are a signed value and therefore this is an arithmetic right shift and the sign bit needs to be propagated. Otherwise, its a logical (or effectively logical) right shift and it doesn't.

Reply:
We only need to track if it's the special sequence that https://github.com/dotnet/runtime/pull/54115 was meant to fix

Reply:
An alternative solution is we can expand the array by 1 element before performing 2's complement and normalize the uint array at the end

Reply:
> We only need to track if it's the special sequence that #54115 was meant to fix
Cuz effectively for that sequence the sign bit is lost after the twoscomplement

Ah, right. We don't store it directly as a negative value, we store the sign + magnitude, which makes this logic more complicated.

Reply:
Can you add a code comment describing why this logic is needed for future readers.  Otherwise this LGTM.

Reply:
Done

Reply:
Oh, i discovered a simpler fix. I should have set the last uint as 0xFFFFFFFF before the 2's complement